python/trifinger_simulation/gym_wrapper/envs/cube_trajectory_env.py

```diff
@@ -0,0 +1,341 @@
+"""Gym environment for the Real Robot Challenge Phase 1 (Simulation)."""
+import enum
+
+import numpy as np
+import gym
+
+from trifinger_simulation import TriFingerPlatform
+from trifinger_simulation import visual_objects
+from trifinger_simulation.tasks import move_cube_on_trajectory as mct
+
+
+class Initializer:
+    """Base initializer for getting a trajectory."""
+
+    def get_trajectory(self) -> mct.Trajectory:
+        raise NotImplementedError()
+
+
+class RandomInitializer(Initializer):
+    """Initializer that samples random trajectories."""
+
+    def get_trajectory(self):
+        """Get a random trajectory."""
+        return mct.sample_trajectory()
+
+
+class FixedInitializer(Initializer):
+    """Initializer that uses a fixed trajectory."""
+
+    def __init__(self, trajectory: mct.Trajectory):
+        """Initialize.
+
+        Args:
+            trajectory: The goal trajectory.
+
+        Raises:
+            Exception:  If trajectory is not valid.  See
+            :meth:`mct.validate_goal` for more information.
+        """
+        mct.validate_trajectory(trajectory)
+        self.trajectory = trajectory
+
+    def get_trajectory(self):
+        """Get the trajectory that was set in the constructor."""
+        return self.trajectory
+
+
+class ActionType(enum.Enum):
+    """Different action types that can be used to control the robot."""
+
+    #: Use pure torque commands.  The action is a list of torques (one per
+    #: joint) in this case.
+    TORQUE = enum.auto()
+    #: Use joint position commands.  The action is a list of angular joint
+    #: positions (one per joint) in this case.  Internally a PD controller is
+    #: executed for each action to determine the torques that are applied to
+    #: the robot.
+    POSITION = enum.auto()
+    #: Use both torque and position commands.  In this case the action is a
+    #: dictionary with keys "torque" and "position" which contain the
+    #: corresponding lists of values (see above).  The torques resulting from
+    #: the position controller are added to the torques in the action before
+    #: applying them to the robot.
+    TORQUE_AND_POSITION = enum.auto()
+
+
+class CubeTrajectoryEnv(gym.GoalEnv):
+    """Gym environment for moving cubes with simulated TriFingerPro."""
+
+    def __init__(
+        self,
+        initializer: Initializer,
+        action_type: ActionType = ActionType.POSITION,
+        step_size: int = 1,
```

Comment:
I think this is also often called the frame skip in RL. But time step together with the doc string should also explain the meaning.